packages/odyssey-react-mui/src/labs/DataTable.tsx

```diff
@@ -0,0 +1,646 @@
+  const [density, setDensity] = useState<MRT_DensityState>(initialDensity);
+  const [columnVisibility, setColumnVisibility] = useState<MRT_VisibilityState>(
+    columns.reduce((acc, column) => {
+      acc[column.accessorKey as string] = true;
+      return acc;
+    }, {} as MRT_VisibilityState)
+  );
+  const [globalFilter, setGlobalFilter] = useState<string>("");
+  const [filters, setFilters] = useState<Array<DataFilter>>();
+
+  const refreshData = async () => {
+    setShowSkeletons(true);
+    try {
+      const newData = await fetchDataFn({
+        page: page,
+        resultsPerPage: resultsPerPage,
+        sort: sorting,
+        search: globalFilter,
+        filters: filters,
+      });
+      setData(newData);
+      setShowSkeletons(false);
+    } catch (error) {
+      console.log(error);
+      setShowSkeletons(false);
+    }
+  };
+
+  const handleSortingChange = (updater: MRT_Updater<MRT_SortingState>) => {
+    setSorting((prevSorting) =>
+      updater instanceof Function ? updater(prevSorting) : sorting
+    );
+  };
+
+  const handleColumnVisibility = (columnId: string) => {
+    setColumnVisibility((prevVisibility) => ({
+      ...prevVisibility,
+      [columnId]: !columnVisibility[columnId],
+    }));
+  };
+
+  const handleSearch = (value: string) => {
+    setGlobalFilter(value);
+  };
+
+  const handleFilters = (updatedFilters: Array<DataFilter>) => {
+    setFilters(updatedFilters);
+  };
+
+  const handleRowSelectionChange = (
+    updater: MRT_Updater<MRT_RowSelectionState>
+  ) => {
+    setRowSelection((prevRowSelection) =>
+      updater instanceof Function ? updater(prevRowSelection) : rowSelection
+    );
+  };
+
+  const handleReordering = ({
+    rowId,
+    newIndex,
+  }: {
+    rowId: string;
+    newIndex: number;
+  }) => {
+    if (newIndex < 0) {
+      return;
+    }
+
+    if (totalRows && newIndex > totalRows) {
+      return;
+    }
+
+    reorderDataFn?.({ rowId: rowId, newIndex: newIndex });
+    refreshData();
+  };
+
+  useEffect(() => {
+    setShowSkeletons(false);
+  }, [data]);
+
+  useEffect(() => {
+    refreshData();
+  }, [page, resultsPerPage, sorting, globalFilter, filters]);
+
+  useEffect(() => {
+    onRowSelectionChange?.(rowSelection);
+  }, [rowSelection, onRowSelectionChange]);
+
+  const rowVirtualizerInstanceRef =
+    useRef<MRT_Virtualizer<HTMLDivElement, HTMLTableRowElement>>(null);
+
+  const table = useMaterialReactTable({
+    columns: columns,
+    data: data,
+    state: {
+      density,
+      sorting,
+      globalFilter,
+      columnVisibility,
+      rowSelection,
+      showSkeletons,
+    },
+    rowVirtualizerInstanceRef: rowVirtualizerInstanceRef,
+    rowVirtualizerOptions: { overscan: 4 },
+    enableRowVirtualization:
+      paginationType === "loadMore" || resultsPerPage > 50,
+    enableColumnResizing: hasColumnResizing,
+    enableDensityToggle: false,
+    enableFullScreenToggle: false,
+    enablePagination: false,
```

Comment:
Do you also need `manualPagination: false` (just like `enableFilters`/`manualFiltering`)?

Reply:
We don't, quite unintuitively, because of the way we're loading in the data — the pagination's being handled by our fetch function.